plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -4,93 +4,112 @@ import (
 	_ "github.com/denisenkom/go-mssqldb" // go-mssqldb initialization
 )
 
+//------------------------------------------------------------------------------------------------
+//------------------ Azure SQL Database ------------------------------------------------------
+//------------------------------------------------------------------------------------------------
 // Only executed if AzureDB flag is set
-const sqlAzureDBResourceStats string = `SET DEADLOCK_PRIORITY -10;
-IF SERVERPROPERTY('EngineEdition') = 5  -- Is this Azure SQL DB?
-BEGIN
-        SELECT TOP(1)
-                'sqlserver_azure_db_resource_stats' AS [measurement],
-                REPLACE(@@SERVERNAME,'\',':') AS [sql_instance],
-                DB_NAME() as [database_name],
-                cast(avg_cpu_percent as float) as avg_cpu_percent,
-                cast(avg_data_io_percent as float) as avg_data_io_percent,
-                cast(avg_log_write_percent as float) as avg_log_write_percent,
-                cast(avg_memory_usage_percent as float) as avg_memory_usage_percent,
-                cast(xtp_storage_percent as float) as xtp_storage_percent,
-                cast(max_worker_percent as float) as max_worker_percent,
-                cast(max_session_percent as float) as max_session_percent,
-                dtu_limit,
-                cast(avg_login_rate_percent as float) as avg_login_rate_percent  ,
-                end_time,
-                cast(avg_instance_memory_percent as float) as avg_instance_memory_percent ,
-                cast(avg_instance_cpu_percent as float) as avg_instance_cpu_percent
-        FROM
-                sys.dm_db_resource_stats WITH (NOLOCK)
-        ORDER BY
-                end_time DESC
+const sqlAzureDBResourceStats string = `
+SET DEADLOCK_PRIORITY -10;
```

Comment:
What is the nead for Deadlock_priority - given DMVs they are memory locations not updatable. I had removed many of them :)... do you see any explicit need for them?

Reply:
my bad, I placed it everywhere when I added the check on the engine edition (I mad the code block once and then pasted it everywhere)

Reply:
I had a look at this, DEADLOCK_PRIORITY was present in 11 queries out of 17 for the Azure part and in 9 out of 9 for the on-prem.
I'll remove it in case queries use only dmv

Reply:
I've removed some DEADLOCK_PRIORITY (where only dmv were referenced), it is now used only in DatabaseIO and PerfCounters queries (both azure DB and MI)